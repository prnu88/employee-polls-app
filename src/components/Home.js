import Poll from "./Poll";
import Nav from "./Nav";
import { connect } from 'react-redux';

function Home(props){    
    const {users,questions} = props;

    const currentUser = users[props.authedUser];
    const answers = currentUser.answers;
    const answeredQuestionsArr = [];
    const unAnsweredQuestionsArr = [];

    for (const key in questions) {
        if( Object.getOwnPropertyNames(answers).includes(key) ){
            answeredQuestionsArr.push(props.questions[key])
        }else{
            unAnsweredQuestionsArr.push(props.questions[key])
        }
    }

    answeredQuestionsArr.sort(
        (a, b) => b.timestamp - a.timestamp
      );

    unAnsweredQuestionsArr.sort(
        (a, b) => b.timestamp - a.timestamp
     );
  
   
    return (<div>
        <Nav></Nav>
        <div className="solidborder">
            <h4>New Questions</h4>
        </div>
        <div  className="solidborder">   
            <div className="parent">
                {unAnsweredQuestionsArr.map((unAnsweredQuestion) => (
                    <div className="child">
                        <Poll key={unAnsweredQuestion.id} id={unAnsweredQuestion.id} author={unAnsweredQuestion.author} timestamp={unAnsweredQuestion.timestamp}/>
                    </div>
                ))}
            </div>
        </div>
        <div><br/></div>
        <div className="solidborder">
            <h4>Done</h4>
        </div>
        <div className="solidborder">
            <div className="parent">
                {answeredQuestionsArr.map((answeredQuestion) => (
                        <div className="child">
                            <Poll  key={answeredQuestion.id}id={answeredQuestion.id} author={answeredQuestion.author} timestamp={answeredQuestion.timestamp}/>
                        </div>
                    ))}
            </div>
        </div>
      
    </div>)
}

const mapStateToProps = ({authedUser,users,questions}) => ({
    loading: authedUser === null,
    authedUser,
    users,
    questions,
  });

export default connect(mapStateToProps)(Home);