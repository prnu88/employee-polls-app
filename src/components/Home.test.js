import Home from './Home';
import { Provider } from "react-redux";
import configureMockStore from "redux-mock-store";

const mockStore = configureMockStore();
const store = mockStore({});

describe("Home Component", () => {
    it("should render without throwing an error", () => {
        expect(
          <Provider store={store}>
          <Home />
          </Provider>
        ).toMatchSnapshot();
    });
});