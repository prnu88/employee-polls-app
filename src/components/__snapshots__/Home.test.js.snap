// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Component should render without throwing an error 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Memo(Connect(Home)) />
</Provider>
`;
